core/trino-main/src/main/java/io/trino/execution/TaskManagerConfig.java

```diff
@@ -98,12 +94,12 @@
     /**
      * default value is overwritten for fault tolerant execution in {@link #applyFaultTolerantExecutionDefaults()}}
      */
-    private int taskConcurrency = min(max(nextPowerOfTwo(getAvailablePhysicalProcessorCount()), 2), 32);
-    private int httpResponseThreads = 100;
-    private int httpTimeoutThreads = 3;
+    private ThreadsCount taskConcurrency = ThreadsCount.boundedValueOf("1C2", "2", "32");
```

Comment:
I think we should just have a number here rather that `ThreadsCount`. It's easier to diagnose system configuration from startup logs then.